Added glossary (first draft) (#6482)
fleetdm/fleet: handbook/brand.md

## Commonly used terms

| Term | Meaning | 
|:------ |:-----------------|
| **antivirus** | A class of programs designed to detect, block, and clear away malware from devices, networks, and IT systems. |
| **API** | (Application Programming Interface) a software go-between that allows applications to communicate.  |
| **automation** | A system that operates without needing intervention from a human to do so. |
| **AWS** | (Amazon Web Services) An ever-evolving cloud computing platform designed to allow application providers, ISVs, and vendors to host applications. |
| **CI/CD** | (Continuous Integration and Continuous Delivery/Continuous Deployment) A software development practice where cumulative code changes are made regularly and accurately. |
| **CLI** | (Command Line Interface) A tool for managing Fleet from the command line. |
| **Client Platform Engineer (CPE)** | See: CPE. |

> | **cloud** | Data storage and access through the internet instead of a device's hard drive. |

Author: ```suggestion
| **cloud** | Data storage, networking, servers, databases, software, intelligence, and analytics through the internet instead of a device's hard drive. |
```

> | **cloud** | Data storage and access through the internet instead of a device's hard drive. |

Author: How is this @GuillaumeRoss?

> | **cloud** | Data storage and access through the internet instead of a device's hard drive. |

Reviewer: Will commit it in a batch for you!

| **command line** | A horizontal row on an interface for text to allow you to type in a variety of commands. Also, see "CLI." |
| **compliance** | The act of being in line with the established risk-based expectations to preserve the strength and confidentiality of data stored, used, and transmitted. |
| **CPE** | (Client Platform Engineer) A person who constructs, evaluates, and deploys solutions to administrate a fleet of "clients" or end-users and does so in a scaleable manner. |

> | **CVE** | (Common Vulnerabilities and Exposures) A system that provides a technique for sharing information publicly. |

Author: ```suggestion
| **CVE** | (Common Vulnerabilities and Exposures) A program that provides a technique for sharing information publicly. |
```

| **data leaks** | When crucial and confidential data is unwittingly exposed physically, on the Internet, or any other way. This includes misplaced hard drives or devices. |
| **device management** | The process of overseeing the execution, process, and upkeep of a device, be it physical or virtual. |
| **DevOps** | Practices that incorporate both software development (Dev) and IT operations (Ops). |
| **Docker** | An open source platform that allows one to manage containerized applications. |
| **DRI** | The person who is singularly responsible for a given aspect of the open source project, the product, or the company. |
| **EDR** | (Endpoint Detection and Response) Security software that continually audits end-user devices to identify and respond to threats such as malware and ransomware. Also, see EDTR. |

> | **EDTR** | (Endpoint Detection and Threat Response) Security software that continually audits end-user devices to identify and respond to threats such as malware and ransomware. Also, see EDR. |

Author: Hey@ Guillaume. If you see a term on this list we've used it somewhere on the website. If we only use it a few times that's all the more reason to have more context.

> | **EDTR** | (Endpoint Detection and Threat Response) Security software that continually audits end-user devices to identify and respond to threats such as malware and ransomware. Also, see EDR. |

Reviewer: I'd love to know where it comes from then because as far as I know, EDTR is not an acronym anyone has ever used 😅

| **encryption** | The act of converting data into a cipher that requires a key to be deciphered. |

> | **end-users** | Someone using a distributed device. This could be a computer or a mobile device. |

Author: ```suggestion
| **end-users** | Someone using a distributed device or service. This could be a computer or a mobile device. |
```

| **Filevault** | A program used on a mac to encrypt data. |
| **Firewall** | A network security device that monitors in and out of your network. |
| **fleetctl** | Can be used to accomplish many tasks you would typically need to do through the UI (User Interface). |

> | **Fleetctl CLI** | (Pronounced "Fleet control") A CLI tool for managing Fleet from the command line. Fleetctl enables a GitOps workflow with Fleet and osquery. |

Author: @GuillaumeRoss, good questions. @noahtalerman, do you have any insight here? I was doing my best from what I found in the docs.

> | **Fleetctl CLI** | (Pronounced "Fleet control") A CLI tool for managing Fleet from the command line. Fleetctl enables a GitOps workflow with Fleet and osquery. |

Reviewer 2: @Desmi-Dizney this is great that you ran into this. Guillaume is correct that `fleetctl` and `fleetctl CLI` are the same.

I think it makes sense to only include `fleetctl`. The definition could become...

> A CLI tool for managing Fleet from the command line. Can be used to accomplish many tasks you would typically need to do through the UI (User Interface). Also, Fleetctl enables a GitOps workflow with Fleet and osquery.

> | **Fleetctl CLI** | (Pronounced "Fleet control") A CLI tool for managing Fleet from the command line. Fleetctl enables a GitOps workflow with Fleet and osquery. |

Reviewer 2: Desmi, as you experienced, the docs are confusing. Maybe at some point we update the docs to only use `fleetctl` naming and not `fleetctl CLI` ?

| **GitHub** | A website that is a cloud-based service to help developers with code management, storage, tracking, and controlling changes. |

> | **historical compliance** | The ability to view past behavior around established risk-based controls to safeguard the integrity, confidentiality, and access of data storage, processing, or transfers. |

Author: As a non-security person, I would not have known if that term was a feature or status. I don't think it hurts to break things down.

| **IETF** | (Internet Engineering Task Force) An organization that defines standardizing operations of internet protocols |
| **Internet Engineering Task Force (IETF)** | See: IETF |
| **IR** | (Incident Response) The actions one takes in response to a security breach or cyberattack. |
| **Linux** | An open source operating system that is also community developed. |

> | **Logica** | An IT and management consultancy company based in the United Kingdom. |

Author: If we refer to it in the docs and the docs don't explain it, the glossary should.

> | **Logica** | An IT and management consultancy company based in the United Kingdom. |

Reviewer: The docs mention Logica? Where?

> | **Logica** | An IT and management consultancy company based in the United Kingdom. |

Author: Sorry @GuillaumeRoss, just like EDTR I'm not sure where in the docs but none of the terms come from outside of Fleet.

| **macOS** | The operating system used in all of Apple's Mac computers. |
| **Munki** | Open-source tools that manage installing and configuring software when used in conjunction with a web server-based repository. |
| **open core** | Is the business model where a company has a core version of a product with some of the features as (FOSS) Free Open Source Software in addition to a paid commercial version that is proprietary software. |
| **open source** | Software with intentionally public code for the sake of transparency. |
| **OS** | (Operating System) Software that provides the groundwork and instructions for a device's basic functions, including application use and controlling peripherals. |
| **osquery**  | A tool that assembles low-level operating system analytics and monitoring. |
| **out-of-policy device** | A device that is fails any security or vulnerability policy created in Fleet. |
| **permissions** | Users have different abilities depending on the access level they have. |

> | **platform** | The primary hardware device and operating system software where applications can be run. |

Author: The most important part is what it means to Fleet so the question is do we use it in these other ways? When is see it in the docs this is the only way I've seen it represented.

> | **platform** | The primary hardware device and operating system software where applications can be run. |

Reviewer: We also use "platform" in the broader sense, like there's a quote from a customer on the first page, in the platform page we mention "data platforms", etc.

> | **platform** | The primary hardware device and operating system software where applications can be run. |

Author: ```suggestion
| **platform** | Any software or hardware for hosting an application, data, or service. |
```

| **policies** | Yes or no questions you can ask using Fleet about your host devices. |
| **policy compliance** | The state of whether a device is passing or failing policies created in Fleet. |
| **queries** | Questions you can ask an end-user device's operating system via Fleet. |
| **SAML** | (Security Assertion Markup Language) A standard that allows identity providers (IdP) to authorize credentials for service providers; enabling SSO (Single Sign-On). |
| **security audits** | A thorough assessment of an organization's data systems. |
| **security engineers** | The responsible individual for the management and up-keep of an organization's security system. |
| **SIEM** | (Security Information and Event Management) Technology that assembles data, security warnings, and events into one platform and provides almost real-time analyzed data to help you better monitor your organization's security. |
| **Site Reliability Engineers (SREs)** | Those who apply scripting and automation to IT jobs such as support and maintenance. |
| **Slack** | A business application for messaging. |

> | **Splunk** | Software that is primarily used for searching, monitoring, and analyzing data in near real-time. |

Author: We can change what it says but it's definitely something we refer to so it at least needs a vague description.

| **SQL** | (Structured Query Language) A language used to manage databases and complete a variety of operations tasks within said databases. |
| **SRE** | See "Site Reliabilty Engineers." |
| **SSO authentication** | (Single Sign-On authentication) Allows identity providers (IdP) to authorize credentials for service providers once and use that as the authentication for multiple outside accounts. |
| **TLS** | (Transport Layer Security) An Internet Engineering Task Force (IETF) standardized protocol that authenticates and provides privacy and data protection over computer networks. |
| **token** | A physical Two-Factor Authentication (2FA) login security device to prove one's identity. |
| **Transport Layer Security (TLS)** | See: TLS |
| **UI** | (User Interface) An interactive space in a program that concentrates on style and intuitive use. |
| **URL** | Uniform resource locator. Specifies where a web resource is located (ex: https://fleetdm.com/articles/) |

> | **vulnerabilities** | An exploitable weakness that can lead to unapproved access to a computer system. |

Reviewer: ```suggestion
| **vulnerabilities** | An exploitable weakness that can lead to unauthorized access or other negative consequences to a computer system. |
```
